### PR TITLE
Us023/delete job family

### PR DESCRIPTION
**Pull request by clarebuckley**

### CODE REVIEW

**Review by sifatj — APPROVED** [review]
Looks good

File: src/app/data.service.ts [hunk 1]
```diff
@@ -58,4 +58,12 @@ export class DataService {
   public getBand(bandId) : Observable<Band>{
     return this.http.get<Band>('/api/bands/' + bandId);
   }
+
+  public getDeletableJobFamilies() : Observable<JobFamily[]> {
+    return this.http.get<JobFamily[]>('/api/deletableJobFamilies');
+  }
+
+  public deleteJobFamily(jobFamilyId): Observable<Object> {
+    return this.http.delete('/api/deleteJobFamily/' + jobFamilyId);
```
Comment (rhysbarrett) [inline src/app/data.service.ts:96]: Probably too late to be worrying about API schemas at this stage, but for future reference, RESTful APIs don't have verbs in the URL. They're unnecessary, as the verb is already in the request by way of the HTTP verb (see your use of `this.http.delete` in this line of code). So, for example, you could have a single route /api/job-family and use different HTTP verbs such as `GET`, `POST`,` PATCH` and `DELETE` to differentiate between the CRUD operations. 

I'm aware of the time constraints so don't worry about this for now, but bear it in mind for when you're building APIs in the future.

File: src/app/data.service.ts [hunk 2]
```diff
@@ -58,4 +58,12 @@ export class DataService {
   public getBand(bandId) : Observable<Band>{
     return this.http.get<Band>('/api/bands/' + bandId);
   }
+
+  public getDeletableJobFamilies() : Observable<JobFamily[]> {
+    return this.http.get<JobFamily[]>('/api/deletableJobFamilies');
```
Comment (rhysbarrett) [inline src/app/data.service.ts:92]: To follow up from my previous comment about RESTful practices, an improvement on this URL would be `api/job-families?deletable=True`

Comment (rhysbarrett) [reply src/app/data.service.ts:92]: But again don't worry about changing this at such a late stage.

**Review by rhysbarrett — APPROVED** [review]
Left some comments for future reference regarding REST

File: src/app/data.service.ts (hunk 1 above)
Comment (clarebuckley) [reply src/app/data.service.ts:96]: I'll keep this in mind for the future, thanks!

**Pull request merged**